src/main/java/sparksoniq/jsoniq/compiler/translator/expr/flowr/OrderByClauseExpr.java

```diff
@@ -48,6 +49,12 @@ public OrderByClauseExpr(
         this._emptyOrder = empty_order;
     }
 
+    @Override
+    public void initHighestExecutionMode() {
+        // OrderByClauseExpr's execution mode is not used. Leave it unset
+        this._highestExecutionMode = ExecutionMode.UNSET;
```

Comment:
Why is it not used, i.e., why is this clause treated differently?

Reply:
This again due to the current nature of how we are creating and using our iterators. These orderbyClauseExpressions are the individual expressions within an order by expression (ie. order by $x asc, $y dec.... ). The execution mode of such individual expressions are not used in the current state of the program. 

Overall orderByClause gets its execution mode set based on the previousClause and this expression's execution mode sets the execution mode of the OrderByClauseSparkIterator. 

The orderByClauseExpr (the subexpressions) contain the actual expression along with the ascending/descending and empty handling information. This actual expression is what the iterator is generated from and it sets the execution mode directly for the iterator. As orderByClauseExpr does not partake in generation and execution mode of an iterator, its execution mode is unused in the current implementation.

I generally opt for setting these UNSET values explicitly in the initialization for showing the deliberate nature of this value, especially for the case if the implementation changes in the future and orderByClauseExpr plays a more direct role for a RuntimeIterator.